core/src/main/java/io/onetable/client/TableFormatClientFactory.java

```diff
@@ -39,16 +37,22 @@ public static TableFormatClientFactory getInstance() {
   }
 
   public TargetClient createForFormat(
-      TableFormat tableFormat, PerTableConfig perTableConfig, Configuration configuration) {
-    switch (tableFormat) {
-      case ICEBERG:
-        return new IcebergClient(perTableConfig, configuration);
-      case DELTA:
-        return new DeltaClient(perTableConfig, configuration);
-      case HUDI:
-        return new HudiTargetClient(perTableConfig, configuration);
-      default:
-        throw new NotSupportedException("Target format is not yet supported: " + tableFormat);
+      String tableFormat, PerTableConfig perTableConfig, Configuration configuration) {
+    TargetClient tc = createTargetClientForName(tableFormat);
+    if (tc == null) {
+      throw new NotSupportedException("Target format is not yet supported: " + tableFormat);
+    }
+    tc.init(perTableConfig, configuration);
+    return tc;
+  }
+
+  public TargetClient createTargetClientForName(String tableFormatName) {
+    ServiceLoader<TargetClient> loader = ServiceLoader.load(TargetClient.class);
+    for (TargetClient target : loader) {
+      if (target.getTableFormat().equalsIgnoreCase(tableFormatName)) {
+        return target;
+      }
     }
+    return null;
```

Comment:
An exception should be thrown and the process should be stopped right away if the configuration has an invalid table format name.